benchadapt/python/benchadapt/_machine_info.py

```diff
@@ -362,6 +388,26 @@ def _has_missing(info, mapping):
     return False
 
 
-def _exec_command(command):
+def _exec_command(command, ignore_error=True):
+    """
+    The ignore_error behavior is for legacy code, it's not advisable to
+    swallow error like this; really hard to debug.
+    """
     result = subprocess.run(command, capture_output=True)
+
+    # Some of these failures are expected, i.e. the command not executing isn't
+    # supposed to be fatal to this program here. However, it makes sense to log
+    # information about how exactly the child process invocation failed.
+    if result.returncode != 0:
+        log.warning(
+            "command `%s` returned with code %s, stderr prefix:\n %s",
+            command,
+            result.returncode,
+            result.stderr.decode("utf-8").strip()[:500],
+        )
+        if not ignore_error:
+            raise Exception(
+                "command unexpectedly returned with non-zero exit code, see log for details"
+            )
```

Comment:
`subprocess.run()` will raise nice classed errors for non-zero return codes [if you set `check=True`](https://docs.python.org/3/library/subprocess.html#subprocess.run) or call [`.check_returncode()`](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.check_returncode) on `result`

Reply:
Oh, yes. I use these all the time. I consider `_machine_info.py` a lost case in terms of error handling, and don't want to spend more seconds in this file than necessary. Me adding a strict-related boolean here was a more or less frustrated response to https://github.com/conbench/conbench/blob/6c6f3a441a07a0e3a4ce3f6f5ca46e0ff3ca8d3e/benchadapt/python/benchadapt/_machine_info.py#L203 

```
def _commands(info):
    for key, command in COMMANDS.items():
        try:
            result = _exec_command(command)
            info[key] = result if result else ""
        except:
            info[key] = ""
```
which -- well, I save my words! :)